### ISSUE TITLE
Crash on start after compiling Gitea 1.16.4 with go 1.18

### ISSUE THREAD

**Issue by ArchangeGabriel** [go-gitea/gitea#19135]
### Gitea Version

1.16.4

### Git Version

2.35.1

### Operating System

Arch Linux

### How are you running Gitea?

Using the Arch Linux package (I am the maintainer), running with systemd ([service file](https://github.com/archlinux/svntogit-community/blob/packages/gitea/trunk/gitea.service)).

### Database

PostgreSQL

### Description

It crashes on start, full log below:
```go
cmd/web.go:102:runWeb() [I] Starting Gitea on PID: 21141
cmd/web.go:150:runWeb() [I] Global init
routers/init.go:106:GlobalInitInstalled() [I] Git Version: 2.35.1, Wire Protocol Version 2 Enabled
routers/init.go:109:GlobalInitInstalled() [I] AppPath: /usr/bin/gitea
routers/init.go:110:GlobalInitInstalled() [I] AppWorkPath: /var/lib/gitea
routers/init.go:111:GlobalInitInstalled() [I] Custom path: /var/lib/gitea/custom
routers/init.go:112:GlobalInitInstalled() [I] Log path: /var/log/gitea/
routers/init.go:113:GlobalInitInstalled() [I] Configuration file: /etc/gitea/app.ini
routers/init.go:114:GlobalInitInstalled() [I] Run Mode: Prod
...dules/setting/log.go:283:newLogService() [I] Gitea v1.16.4 built with GNU Make 4.3, go1.18 : bindata, sqlite, sqlite_unlock_notify, pam
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
...eful/manager_unix.go:144:handleSignals() [W] PID: 21141. Background context for manager closed - context canceled - Shutting down...
cmd/web.go:91:func1() [F] PANIC: runtime error: invalid memory address or nil pointer dereference
        runtime/panic.go:220 (0x2c0a65a8c35)
        runtime/signal_unix.go:818 (0x2c0a65a8c05)
        runtime/alg.go:352 (0x2c0a65666b8)
        runtime/hash64.go:91 (0x2c0a6566728)
        runtime/hash64.go:53 (0x2c0a65666bb)
        runtime/alg.go:54 (0x2c0a655ea44)
        runtime/map.go:934 (0x2c0a656c26a)
        runtime/map.go:861 (0x2c0a656bf47)
        runtime/map.go:1373 (0x2c0a65bf5d8)
        github.com/modern-go/reflect2@v1.0.1/unsafe_map.go:112 (0x2c0a6752ce4)
        github.com/json-iterator/go@v1.1.11/reflect_map.go:291 (0x2c0a6752b8b)
        github.com/json-iterator/go@v1.1.11/reflect.go:219 (0x2c0a674afe1)
        github.com/json-iterator/go@v1.1.11/reflect.go:98 (0x2c0a674a2f7)
        github.com/json-iterator/go@v1.1.11/config.go:299 (0x2c0a6741408)
        code.gitea.io/gitea/modules/json/json.go:78 (0x2c0a676bdf6)
        code.gitea.io/gitea/modules/json/json.go:103 (0x2c0a69f0c48)
        code.gitea.io/gitea/modules/setting/log.go:205 (0x2c0a69f0c2c)
        code.gitea.io/gitea/modules/setting/log.go:323 (0x2c0a69f22d4)
        code.gitea.io/gitea/modules/setting/log.go:349 (0x2c0a69f271c)
        code.gitea.io/gitea/modules/setting/setting.go:1224 (0x2c0a6a02090)
        code.gitea.io/gitea/routers/init.go:74 (0x2c0a8154dd8)
        code.gitea.io/gitea/routers/init.go:119 (0x2c0a8155404)
        code.gitea.io/gitea/cmd/web.go:153 (0x2c0a8230130)
        github.com/urfave/cli@v1.22.5/app.go:524 (0x2c0a7bfe4a7)
        github.com/urfave/cli@v1.22.5/command.go:173 (0x2c0a7bff291)
        github.com/urfave/cli@v1.22.5/app.go:277 (0x2c0a7bfbb06)
        code.gitea.io/gitea/main.go:115 (0x2c0a82662d8)
        runtime/proc.go:250 (0x2c0a6594f12)
        runtime/asm_amd64.s:1571 (0x2c0a65c5b40)
```

**Comment by jcpetro97** [comment]
I am seeing this when building a docker image ( arm7 and X86) as well.  Exact same error as in the output already provided, except I am using sqlite instead of PostgreSQL.

**Comment by ArchangeGabriel** [comment]
FWIW I don’t think the DB has any impact at this stage.

**Comment by ArchangeGabriel** [comment]
Fixed by backporting #19100 as in #19134.
